src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -1234,48 +1236,68 @@ public static string GetDecoratedSymbolName(SymbolReference symbolReference)
             await requestContext.SendResult(result);
         }
 
+        // Since the NamedPipeConnectionInfo type is only available in 5.1+
+        // we have to use Activator to support older version of PS.
+        // This code only lives in the v1.X of the extension.
+        // The 2.x version of the code can be found here:
+        // https://github.com/PowerShell/PowerShellEditorServices/pull/881
+        private static Type _namedPipeConnectionInfoType = Type.GetType("System.Management.Automation.Runspaces.NamedPipeConnectionInfo, System.Management.Automation");
+        private static ConstructorInfo _namedPipeConnectionInfoCtor = _namedPipeConnectionInfoType.GetConstructor(new [] { typeof(int) });
+        private static Runspace GetRemoteRunspace(int pid)
+        {
+            var namedPipeConnectionInfoInstance = _namedPipeConnectionInfoCtor.Invoke(new object[] { pid });
+            return RunspaceFactory.CreateRunspace(namedPipeConnectionInfoInstance as RunspaceConnectionInfo);
+        }
+
         protected async Task HandleGetRunspaceRequestAsync(
             string processId,
             RequestContext<GetRunspaceResponse[]> requestContext)
         {
-            var runspaceResponses = new List<GetRunspaceResponse>();
+            IEnumerable<PSObject> runspaces = null;
 
             if (this.editorSession.PowerShellContext.LocalPowerShellVersion.Version.Major >= 5)
             {
                 if (processId == null) {
                     processId = "current";
                 }
 
-                var isNotCurrentProcess = processId != null && processId != "current";
-
-                var psCommand = new PSCommand();
-
-                if (isNotCurrentProcess) {
-                    psCommand.AddCommand("Enter-PSHostProcess").AddParameter("Id", processId).AddStatement();
-                }
-
-                psCommand.AddCommand("Get-Runspace");
-
-                StringBuilder sb = new StringBuilder();
-                IEnumerable<Runspace> runspaces = await editorSession.PowerShellContext.ExecuteCommand<Runspace>(psCommand, sb);
-                if (runspaces != null)
+                // If the processId is a valid int, we need to run Get-Runspace within that process
+                // otherwise just use the current runspace.
+                if (int.TryParse(processId, out int pid))
                 {
-                    foreach (var p in runspaces)
+
+                    // Create a remote runspace that we will invoke Get-Runspace in.
+                    using(var rs = GetRemoteRunspace(pid))
```

Comment:
```suggestion
                    using(Runspace rs = GetRemoteRunspace(pid))
```

Optional, just skirting the line on the obviousness of the return value here.

Reply:
fixed